snuba/web/rpc/v1/endpoint_get_traces.py

```diff
@@ -0,0 +1,289 @@
+from snuba.datasets.entities.entity_key import EntityKey
+from snuba.datasets.entities.factory import get_entity
+from snuba.datasets.pluggable_dataset import PluggableDataset
+from snuba.query import OrderBy, OrderByDirection, SelectedExpression
+from snuba.query.data_source.simple import Entity
+from snuba.query.dsl import Functions as f
+from snuba.query.dsl import column
+from snuba.query.expressions import Expression
+from snuba.query.logical import Query
+from snuba.query.query_settings import HTTPQuerySettings
+from snuba.request import Request as SnubaRequest
+from snuba.web.query import run_query
+from snuba.web.rpc import RPCEndpoint
+from snuba.web.rpc.common.common import (
+    base_conditions_and,
+    trace_item_filters_to_expression,
+    treeify_or_and_conditions,
+)
+from snuba.web.rpc.common.debug_info import (
+    extract_response_meta,
+    setup_trace_query_settings,
+)
+from snuba.web.rpc.common.exceptions import BadSnubaRPCRequestException
+
+_DEFAULT_ROW_LIMIT = 10_000
+
+_COLUMN_TO_NAME: dict[TraceColumn.Name, str] = {
+    TraceColumn.Name.NAME_FILTERED_SPAN_COUNT: "filtered_span_count",
+    TraceColumn.Name.NAME_ROOT_SPAN_NAME: "root_span_name",
+    TraceColumn.Name.NAME_START_TIMESTAMP: "start_timestamp",
+    TraceColumn.Name.NAME_TOTAL_SPAN_COUNT: "total_span_count",
+    TraceColumn.Name.NAME_TRACE_ID: "trace_id",
+}
+
+_NAME_TO_COLUMN: dict[str, TraceColumn.Name] = {
+    v: k for k, v in _COLUMN_TO_NAME.items()
+}
+
+_TYPES_TO_CLICKHOUSE: dict[AttributeKey.Type, str] = {
+    AttributeKey.Type.TYPE_STRING: "String",
+    AttributeKey.Type.TYPE_INT: "Int64",
+    AttributeKey.Type.TYPE_FLOAT: "Float64",
+}
+
+_POSSIBLE_TYPES: dict[TraceColumn.Name, set[AttributeKey.Type]] = {
+    TraceColumn.Name.NAME_TRACE_ID: {
+        AttributeKey.Type.TYPE_STRING,
+    },
+    TraceColumn.Name.NAME_START_TIMESTAMP: {
+        AttributeKey.Type.TYPE_STRING,
+        AttributeKey.Type.TYPE_INT,
+        AttributeKey.Type.TYPE_FLOAT,
+    },
+}
+
+
+def _column_to_expression(trace_column: TraceColumn, conditions=None) -> Expression:
+    if trace_column.name == TraceColumn.Name.NAME_TOTAL_SPAN_COUNT:
+        return f.count(
+            alias=_COLUMN_TO_NAME[trace_column.name],
+        )
+    if trace_column.name == TraceColumn.Name.NAME_FILTERED_SPAN_COUNT:
+        return f.countIf(
+            conditions,
+            alias=_COLUMN_TO_NAME[trace_column.name],
+        )
+    if trace_column.name == TraceColumn.Name.NAME_START_TIMESTAMP:
+        return f.CAST(
+            f.min(column("start_timestamp")),
+            _TYPES_TO_CLICKHOUSE[trace_column.type],
+            alias=_COLUMN_TO_NAME[trace_column.name],
+        )
+    if trace_column.name == TraceColumn.Name.NAME_ROOT_SPAN_NAME:
+        return f.anyIf(
+            column("name"),
+            f.equals(column("is_segment"), True),
+            alias=_COLUMN_TO_NAME[trace_column.name],
+        )
```

Comment:
This has a chance of returning `null` due to our data model. Can the first option we add be the name of the first span which is guaranteed to exist and we add the rest as needed?

So something like `argMin(name, start_timestamp)`